corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -328,10 +325,102 @@ hqDefine("geospatial/js/case_grouping_map",[
             (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16),
         );
     }
+
     function caseGroupSelectModel() {
         'use strict';
         var self = {};
 
+        self.allGroups = ko.observableArray([]);
+        self.allCaseGroups;
+        self.showGroupSelectionTable = ko.computed(function() {
+            return groupLockModelInstance.groupsLocked();
+        })
+        self.visibleGroupIDs = ko.observableArray([]);
+
+        self.groupIDInVisibleGroupIds = function(groupID) {
+            if (self.visibleGroupIDs().indexOf(groupID) !== -1) {
+                return true;
+              } else {
+                return false;
+              }
+        };
+
+        self.getGroupByID = function(groupID) {
+            return _.filter(self.allGroups(), function(group) {return group.groupID === groupID})[0];
```

Comment:
The `_.filter()` function will loop through the entire array regardless of when a result is found. You can use `_.find()` instead to exit the loop and return a result as soon as it is found.

Reply:
Ooh nice, thanks

Reply:
https://github.com/dimagi/commcare-hq/pull/33680/commits/b514f9e1390c71bc57c0e914b872458f0532afa6